src/N98/Magento/Command/Developer/Asset/ClearCommand.php

```diff
@@ -0,0 +1,208 @@
+    {
+        $theme = '/' . trim($theme, '/');
+        $themeLength = strlen($theme);
+
+        /** @var Read $dir */
+        $dir = $this->filesystem->getDirectoryRead($code);
+        $entries = $dir->readRecursively('');
+        $paths = [];
+        foreach ($entries as $entry) {
+            if (substr($entry, -$themeLength) === $theme &&
+                $dir->isDirectory($entry)) {
+                $paths[] = $entry;
+            }
+        }
+
+        return $paths;
+    }
+
+    /**
+     * @param WriteInterface $dir
+     * @param string $path
+     * @return void
+     */
+    private function deleteDirectory($dir, $path)
+    {
+        try {
+            $dir->delete($path);
+            $this->messages[] = '<info><comment>' . $dir->getAbsolutePath() . $path . '</comment> deleted</info>';
+        } catch (\Exception $e) {
+            $this->messages[] = '<error>' . $e->getMessage() . '</error>';
+            if ($this->output->getVerbosity() >= OutputInterface::VERBOSITY_VERBOSE) {
+                $this->messages[] = '<debug>' . $e->getTraceAsString() . '</debug>';
+            }
+        }
+    }
+
+    /**
+     * @param string $code
+     * @return void
+     */
+    private function emptyDirectory($code)
+    {
+        /** @var WriteInterface $dir */
+        $dir = $this->getDirectoryWrite($code);
+        if ($dir) {
+            foreach ($dir->search('*') as $path) {
+                if ($path !== '.' && $path !== '..') {
+                    $this->deleteDirectory($dir, $path);
+                }
+            }
+        }
+    }
+
+    /**
+     * @param string $theme
+     * @param string $code
+     * @return void
+     */
+    private function deleteThemeDirectories($theme, $code)
+    {
+        /** @var WriteInterface $dir */
+        $dir = $this->getDirectoryWrite($code);
+        if ($dir) {
+            $paths = $this->findThemePaths($theme, $code);
+            foreach ($paths as $path) {
+                $this->deleteDirectory($dir, $path);
+            }
+        }
+    }
+
+    /**
+     * @param array $themes
+     * @return void
+     */
+    private function clearThemes($themes)
+    {
+        foreach ($themes as $theme) {
+            $this->deleteThemeDirectories($theme, DirectoryList::STATIC_VIEW);
+            $this->deleteThemeDirectories($theme, DirectoryList::TMP_MATERIALIZATION_DIR);
+        }
+    }
+
+    /**
+     * @return void
+     */
+    private function clearAllThemes()
+    {
+        $this->emptyDirectory(DirectoryList::STATIC_VIEW);
+        $this->emptyDirectory(DirectoryList::TMP_MATERIALIZATION_DIR);
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->detectMagento($output, true);
+        if (!$this->initMagento()) {
+            return;
+        }
+        $this->output = $output;
```

Comment:
This needs a check that prevents running this command in production mode. This is for the users safety if they don't have production on a read-only file-system for the assets which can not be assumed. Then deleting the assets in error would require to re-generate them which can take quite some time.